matlab/+caffe/private/caffe_.cpp

```diff
@@ -586,6 +586,41 @@ static handler_registry handlers[] = {
 };
 
 /** -----------------------------------------------------------------
+** failure handling.
+**/
+
+// Failure function to prevent Matlab from crushing
+// by throwing an exception
+// Failurefunction might be called multiple times, therefore set a first-call variable
+bool firstFail = true;
+void MyFailureFunction(){
+	if (Caffe::using_Matlab() && firstFail){
+		firstFail = false;
+		mexPrintf("failure function\n");
+		throw ("Fatal Error");
```

Comment:
I think using the matlab error funcitons would make more sense here since the google C++ style guide forbids exceptions. Also the caffe code based was not really designed to be recovered from when a `CHECK` fails so this might have some unexpected side effects.